packages/cloud_firestore/ios/Classes/CloudFirestorePlugin.m

```diff
@@ -112,7 +114,93 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
   void (^defaultCompletionBlock)(NSError *) = ^(NSError *error) {
     result(error.flutterError);
   };
-  if ([@"DocumentReference#setData" isEqualToString:call.method]) {
+  if ([@"Firestore#runTransaction" isEqualToString:call.method]) {
+    [[FIRFirestore firestore] runTransactionWithBlock:^id(FIRTransaction *transaction,
+                                                          NSError **pError) {
+      NSNumber *transactionId = call.arguments[@"transactionId"];
+      NSNumber *transactionTimeout = call.arguments[@"transactionTimeout"];
+
+      self.transactions[transactionId] = transaction;
+
+      dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+
+      [self.channel invokeMethod:@"DoTransaction"
+                       arguments:call.arguments
+                          result:^(id doTransactionResult) {
+                            self.transactionResults[transactionId] = doTransactionResult;
+                            dispatch_semaphore_signal(semaphore);
+                          }];
+
+      dispatch_semaphore_wait(
+          semaphore, dispatch_time(DISPATCH_TIME_NOW, [transactionTimeout integerValue] * 1000000));
+
+      return self.transactionResults[transactionId];
+    }
+        completion:^(id transactionResult, NSError *error) {
+          if (error != nil) {
+            NSLog(@"This is the transaction error: %@", error.localizedDescription);
+            result([FlutterError errorWithCode:[NSString stringWithFormat:@"%ld", error.code]
+                                       message:error.localizedDescription
+                                       details:nil]);
+          }
+          result(transactionResult);
+        }];
+  } else if ([@"Transaction#get" isEqualToString:call.method]) {
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+      NSNumber *transactionId = call.arguments[@"transactionId"];
+      NSString *path = call.arguments[@"path"];
+      FIRDocumentReference *documentReference = [[FIRFirestore firestore] documentWithPath:path];
+      FIRTransaction *transaction = self.transactions[transactionId];
+      NSError *pError = [[NSError alloc] init];
```

Comment:
I think this naming is confusing, it should just be `error`. `pError` would be for a `NSError **`

Reply:
Done!